tests/test_data.py

```diff
@@ -0,0 +1,24 @@
+import pytest
+from ml_tooling.utils import DataSetError
+
+
+def test_sqldataset_works_correctly(test_sqldata):
+    assert test_sqldata._x is None
+    features = test_sqldata.x
+    assert test_sqldata._x is not None
```

Comment:
Same test twice? Maybe test _y?

Reply:
It's testing the property works - `._x` should start out as None, but if `.x` is accessed, it should populate `._x` with data, i.e not be None